control-plane-operator/controllers/hostedcontrolplane/oauth/idp_convert.go

```diff
@@ -596,37 +668,93 @@ func isValidURL(rawurl string, optional bool) bool {
 	return u.Scheme == "https" && len(u.Host) > 0 && len(u.Fragment) == 0
 }
 
-func transportForCARef(ctx context.Context, kclient crclient.Client, namespace, name, key string) (http.RoundTripper, error) {
+func transportForCARef(ctx context.Context, kclient crclient.Client, namespace, caName, caKey string, skipKonnectivityDialer bool) (http.RoundTripper, error) {
+	var konnectivityDialer konnectivityproxy.ProxyDialer
+	var userProxyConfig *httpproxy.Config
+	var userProxyTrustedCA string
+
 	// copy default transport
 	transport := net.SetTransportDefaults(&http.Transport{
 		TLSClientConfig: &tls.Config{},
 	})
+	roots := x509.NewCertPool()
+
+	if !skipKonnectivityDialer {
+		var err error
+		// Build dialer for konnectivity.
+		konnectivityDialer, err = buildKonnectivityDialer(ctx, kclient, namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build konnectivity dialer: %w", err)
+		}
+
+		// Fetch user Proxy info.
+		hcpList := &hyperv1.HostedControlPlaneList{}
+		if err := kclient.List(ctx, hcpList); err != nil {
+			return nil, fmt.Errorf("failed to get hosted control plane list: %w", err)
+		}
+		if len(hcpList.Items) != 1 {
+			return nil, fmt.Errorf("expected one hosted control plane, got %d", len(hcpList.Items))
+		}
+		hcp := hcpList.Items[0]
+		if proxy := hcp.Spec.Configuration.Proxy; proxy != nil {
+			userProxyConfig = &httpproxy.Config{
+				HTTPProxy:  proxy.HTTPProxy,
+				HTTPSProxy: proxy.HTTPSProxy,
+				NoProxy:    proxy.NoProxy,
```

Comment:
We should call https://github.com/openshift/hypershift/blob/ffe665196be2e2654d47fa3c798330625cdf493d/support/proxy/no_proxy.go#L15 to get the value for this.

Reply:
addressed!